app.rb

```diff
@@ -0,0 +1,194 @@
+      '5': 'Create a rental',
+      '6': 'List all rentals for a given person id',
+      '7': 'Exit'
+    }
+
+    # List of books
+    @books = []
+
+    # List of people
+    @people = []
+  end
+
+  # display commands
+  def display_command
+    @command.each do |index, command|
+      puts "#{index} - #{command}"
+    end
+  end
+
+  # Print out all books
+  def list_all_books
+    if @books.empty?
+      puts 'Please insert books first!!'
+    else
+      @books.each_with_index do |book, index|
+        puts "#{index}) Title: #{book.title}, Author: #{book.author}}"
+      end
+    end
+  end
+
+  # Print out all books
+  def list_all_people
+    if @people.empty?
+      puts 'Please insert people first!!'
+    else
+      @people.each do |person|
+        puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+      end
+    end
+  end
+
+  def list_all_person_with_numbers
+    @people.each_with_index do |person, index|
+      puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+    end
+  end
+
+  # Create Student
+  def create_student
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Has parent permission? [Y/N]: '
+    permission = gets.chomp.downcase
+    case permission
+    when 'y' then parent_permission = true
+    when 'n' then parent_permission = false
+    end
+
+    # Create a student and push it to people array
+    @people.push(Student.new('11', age, name, parent_permission: parent_permission))
+  end
+
+  # Create teacher
+  def create_teacher
+    print 'Age: '
+    age = gets.chomp
+
+    print 'Name: '
+    name = gets.chomp
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    # Create a student and push it to people array
+    @people.push(Teacher.new(specialization, age, name))
+  end
+
+  # Create a person
+  def create_person
+    print 'Do you want to create a student (1) or a teacher (2)? [Input the nummber]: '
+    case gets.chomp
+    when '1'
+      create_student
+
+    when '2'
+      create_teacher
+    end
+
+    # Success message
+    puts 'Person Created successfully'
+  end
+
+  # Create a Book
+  def create_book
+    print 'Title: '
+    title = gets.chomp
+
+    print 'Author: '
+    author = gets.chomp
+
+    # Create the book object and add it to the books list
+    @books.push(Book.new(title, author))
+
+    # Success message
+    puts 'Book Created successfully'
+  end
+
+  # Create a rental
+  def create_rental()
+    # Select Book
+    puts 'Select a book from the following list by number'
+    list_all_books
+    book_option = gets.chomp.to_i
+    puts
+
+    # Select Person
+    puts 'Select a person from the following list by number (not id)'
+    list_all_person_with_numbers
+    person_option = gets.chomp.to_i
+    puts
+
+    # Select Date
+    print 'Date: '
+    date = gets.chomp
+
+    # Create the book object and add it to the books list
```

Comment:
Good job working on the project. But whenever I enter the date when trying to create a rental Im getting an error 

This is the error Im getting:

![oop snip](https://user-images.githubusercontent.com/79506105/210864364-84151e8f-38c8-4a80-b8cf-69f57b234fce.PNG)
It would be nice if you fixed that error

Reply:
@KayLemba Thanks a lot for your correction. I tried to fix that issue but find that the good format of date is to start by Year/ Month then Day. In our case 2023/01/03. According to the video, in that way, it is working perfectly

Reply:
Hi @KayLemba 
Thank you very much for meticulously reviewing my code. It actually helped me to understand it better.
I think I have already fixed the issue 😃